File: backend/src/main/java/com/career/plan/service/ProgressService.java
package com.career.plan.service;

import com.career.plan.entity.Progress;
import com.career.plan.repository.ProgressRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Service
public class ProgressService {
    
    @Autowired
    private ProgressRepository progressRepository;
    
    public List<Progress> getAllProgress() {
        return progressRepository.findAll();
    }
    
    public List<Progress> getByPlanId(Long planId) {
        return progressRepository.findByPlanId(planId);
    }
    
    public List<Progress> getByTaskId(Long taskId) {
        return progressRepository.findByTaskId(taskId);
    }
    
    public Progress createProgress(Progress progress) {
        return progressRepository.save(progress);
    }
    
    public Progress updateProgress(Long id, Progress progress) {
        progress.setId(id);
        return progressRepository.save(progress);
    }
    
    public void deleteProgress(Long id) {
        progressRepository.deleteById(id);
    }
    
    public Map<String, Object> getProgressSummary() {
        List<Progress> allProgress = progressRepository.findAll();
        
        Map<String, Object> summary = new HashMap<>();
        summary.put("total", allProgress.size());
        summary.put("reports", allProgress);
        
        return summary;
    }
    
    /**
     * 获取所有计划的进度历史（用于图表展示）
     * @return 所有进度记录
     */
    public List<Progress> getAllProgressHistory() {
        return progressRepository.findAll();
    }
}

File: frontend/src/services/ProgressService.js
import api from './api';

/**
 * 进度服务
 * 处理进度相关的 API 调用
 */
class ProgressService {
  /**
   * 获取计划进度
   * @param {number} planId - 计划 ID
   * @returns {Promise<Object>} 进度数据
   */
  async getPlanProgress(planId) {
    return await api.get(`/progress/plan/${planId}`);
  }

  /**
   * 获取进度摘要
   * @returns {Promise<Object>} 进度摘要
   */
  async getProgressSummary() {
    return await api.get('/progress/summary');
  }

  /**
   * 获取进度历史
   * @param {number} planId - 计划 ID
   * @returns {Promise<Array>} 进度历史记录
   */
  async getProgressHistory(planId) {
    return await api.get(`/progress/history/${planId}`);
  }

  /**
   * 创建进度报告
   * @param {Object} progressData - 进度数据
   * @returns {Promise<Object>} 创建的进度报告
   */
  async createProgress(progressData) {
    return await api.post('/progress-reports', progressData);
  }

  /**
   * 更新进度报告
   * @param {number} id - 进度报告 ID
   * @param {Object} progressData - 进度数据
   * @returns {Promise<Object>} 更新后的进度报告
   */
  async updateProgress(id, progressData) {
    return await api.put(`/progress-reports/${id}`, progressData);
  }
}

export default new ProgressService();
